osu.Game.Modes.Osu/Objects/Slider.cs

```diff
@@ -53,11 +54,13 @@ public override void SetDefaultsFromBeatmap(Beatmap beatmap)
 
             var baseDifficulty = beatmap.BeatmapInfo.BaseDifficulty;
 
-            var startBeatLength = beatmap.BeatLengthAt(StartTime);
-            var multipliedStartBeatLength = beatmap.BeatLengthAt(StartTime, true);
+            ControlPoint overridePoint;
+            ControlPoint timingPoint = beatmap.TimingPointAt(StartTime, out overridePoint);
+            var velocityAdjustment = overridePoint?.VelocityAdjustment ?? 1;
+            var baseVelocity = 100 * baseDifficulty.SliderMultiplier;
 
-            Velocity = 100 / multipliedStartBeatLength * baseDifficulty.SliderMultiplier;
-            TickDistance = (100 * baseDifficulty.SliderMultiplier) / baseDifficulty.SliderTickRate / (multipliedStartBeatLength / startBeatLength);
+            Velocity = baseVelocity / (timingPoint.BeatLength * velocityAdjustment);

```

Comment:
Can divide `baseVelocity` by `velocityAdjustment`. Then you don't need to divide by it twice when setting `Velocity` and `TickDistance`.